Update developing_collections_distributing.rst (#74820)
ansible/ansible: docs/docsite/rst/dev_guide/developing_collections_distributing.rst

The ``ansible-galaxy collection publish`` command triggers an import process, just as if you uploaded the collection through the Galaxy website. The command waits until the import process completes before reporting the status back. If you want to continue without waiting for the import result, use the ``--no-wait`` argument and manually look at the import progress in your `My Imports <https://galaxy.ansible.com/my-imports/>`_ page.

> Alternatively tt is also possible to publish to Galaxy without configuring the token in :file:`ansible.cfg`.  Use the ``--token`` argument directly on the command line.

Reviewer: ```suggestion
Alternatively, it is also possible to publish to Galaxy without configuring the token in :file:`ansible.cfg`.  Use the ``--token`` argument directly on the command line.
```

> Alternatively, it is also possible to publish to Galaxy without configuring the token in :file:`ansible.cfg`.  Use the ``--token`` argument directly on the command line.

Reviewer 2: ```suggestion
Alternately, you can publish to Galaxy without configuring the token in :file:`ansible.cfg`.  Use the ``--token`` argument directly on the command line.
```

.. code-block:: bash

>      ansible-galaxy collection publish path/to/my_namespace-my_collection-1.0.0.tar.gz --token abcdefghijklmnopqrtuvwxyz

Reviewer: Do you think we should advertise this since passing secrets on cmdline may lead to exposing them to others on the system? or you might want to add a line saying that "this is the available but not recommended method" or something...

Reviewer: btw, https://docs.ansible.com/ansible/devel/dev_guide/developing_collections_distributing.html#specifying-your-api-token-with-the-token-argument discusses this.

Author: In general I don't think obfuscating or "not advertising" the way the tool works is a great idea.  If we truly believe that no one should use the `--token` feature we should remove it from the code base.  I would assume that tons of employees are using this feature from their laptops as it speeds up development and testing.  I don't mind adding some more info here about security, but I imagine this is like every other tool out there where tokens are used on the command line.

Reviewer: Agree. While you are here, can you add that note then?

Author: @Akasurde maybe link to that as well... (e.g. link there for another example https://docs.ansible.com/ansible/devel/dev_guide/developing_collections_distributing.html#specifying-your-api-token)

Reviewer 2: @Akasurde - I'm working on a separate PR to add anchors in that file. Once I have that, all include the link above as well. I think Sean has had enough rst fun for one day ;-)
